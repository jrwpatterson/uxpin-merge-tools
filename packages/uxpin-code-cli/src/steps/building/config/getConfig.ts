// tslint:disable-next-line:import-name
import HappyPack = require('happypack');
import { join } from 'path';
import { Configuration } from 'webpack';
import { smart } from 'webpack-merge';

export const TEMP_DIR_PATH:string = './.uxpin-temp';
export const LIBRARY_INPUT_PATH:string = `${TEMP_DIR_PATH}/components.js`;
export const LIBRARY_OUTPUT_PATH:string = `${TEMP_DIR_PATH}/designsystemlibrary.js`;

export function getConfig(projectRoot:string, webpackConfigPath?:string):Configuration {
  const config:Configuration = {
    entry: LIBRARY_INPUT_PATH,
    module: {
      rules: [
        {
          exclude: /node_modules/,
          test: /.jsx?$/,
          use: 'happypack/loader',
        },
      ],
    },
    output: {
      filename: LIBRARY_OUTPUT_PATH,
      libraryTarget: 'commonjs',
    },
    plugins: [
      new HappyPack({
        loaders: [
          {
            loader: 'babel-loader',
            options: {
              plugins: [
                'transform-class-properties',
                'transform-object-rest-spread',
              ],
              presets: [
                ['env', {
                  targets: {
                    browsers: ['last 2 versions'],
                    node: 'current',
                  },
                }],
                'react',
              ],
            },
          },
        ],
      }),
    ],
    resolve: {
      extensions: ['.js', '.jsx'],
    },
  };

  if (webpackConfigPath) {
    const userWebpackConfig:Configuration = require(join(projectRoot, webpackConfigPath));
    return smart(userWebpackConfig, config);
  }
  return config;
}
