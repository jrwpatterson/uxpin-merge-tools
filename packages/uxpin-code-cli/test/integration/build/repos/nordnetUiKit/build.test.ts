import { relative, resolve } from 'path';

import { LIBRARY_OUTPUT_PATH } from '../../../../../src/steps/building/config/getConfig';
import { runUXPinCodeCommand } from '../../../../utils/command/runUXPinCodeCommand';
import { setTimeoutBeforeAll } from '../../../../utils/command/setTimeoutBeforeAll';

const CURRENT_TIMEOUT:number = 75000;

setTimeoutBeforeAll(CURRENT_TIMEOUT);

describe('Building repos/nordnet-ui-kit design system', () => {
  describe('with required user webpack config', () => {
    let components:any;
    let consoleOutput:string;

    beforeAll(() => {
      const params:string[] = [
        '--webpack-config "../../configs/nordnet-ui-kit-webpack.config.js"',
        '--config="../../configs/nordnet-ui-kit-uxpin.config.js"',
        '--wrapper "documentation/wrapper.jsx"',
      ];

      return runUXPinCodeCommand({ cwd: 'resources/repos/nordnet-ui-kit', params })
        .then((output) => {
          const path:string = relative(__dirname, resolve('test/resources/repos/nordnet-ui-kit', LIBRARY_OUTPUT_PATH));
          components = require(path);
          consoleOutput = output;
        });
    });

    it('contains Button component', () => {
      // when
      const { Button } = components;
      // then
      expect(Button).toBeInstanceOf(Function);
    });

    it('contains custom wrapper', () => {
      // when
      const { Wrapper } = components;
      // then
      expect(Wrapper).toBeInstanceOf(Function);
    });

    it('prints warnings without stack traces to the console', () => {
      // then
      expect(consoleOutput).toMatchSnapshot();
    });
  });

  describe('without required user webpack config', () => {
    it('throws an error', () => {
      return runUXPinCodeCommand({ cwd: 'resources/repos/nordnet-ui-kit' })
        .then((output) => {
          expect(output).toContain('ERROR:');
          expect(output).toContain('Module parse failed');
        });
    });
  });
});
