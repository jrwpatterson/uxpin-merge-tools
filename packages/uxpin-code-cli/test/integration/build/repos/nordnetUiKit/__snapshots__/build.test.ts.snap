// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Building repos/nordnet-ui-kit design system with required user webpack config prints warnings without stack traces to the console 1`] = `
"Alert
    ğŸ“œ documentation: âœ”
    ğŸ’¡ examples: âœ˜

Animate
    ğŸ“œ documentation: âœ”
    ğŸ’¡ examples: âœ˜

Avatar
    ğŸ“œ documentation: âœ”
    ğŸ’¡ examples: âœ˜

Badge
    ğŸ“œ documentation: âœ”
    ğŸ’¡ examples: âœ˜

Button
    ğŸ“œ documentation: âœ”
    ğŸ’¡ examples: âœ˜

Dropdown
    ğŸ“œ documentation: âœ”
    ğŸ’¡ examples: âœ˜

Flag
    ğŸ“œ documentation: âœ”
    ğŸ’¡ examples: âœ˜

Icon
    ğŸ“œ documentation: âœ”
    ğŸ’¡ examples: âœ˜

Input
    ğŸ“œ documentation: âœ”
    ğŸ’¡ examples: âœ˜

InstrumentBadge
    ğŸ“œ documentation: âœ”
    ğŸ’¡ examples: âœ˜

LabeledValue
    ğŸ“œ documentation: âœ”
    ğŸ’¡ examples: âœ˜

Li
    ğŸ“œ documentation: âœ”
    ğŸ’¡ examples: âœ˜

Logo
    ğŸ“œ documentation: âœ”
    ğŸ’¡ examples: âœ˜

Pane
    ğŸ“œ documentation: âœ”
    ğŸ’¡ examples: âœ˜

ProgressBar
    ğŸ“œ documentation: âœ”
    ğŸ’¡ examples: âœ˜

RadioGroup
    ğŸ“œ documentation: âœ”
    ğŸ’¡ examples: âœ˜

SegmentedControl
    ğŸ“œ documentation: âœ”
    ğŸ’¡ examples: âœ˜

SharevilleLogo
    ğŸ“œ documentation: âœ”
    ğŸ’¡ examples: âœ˜

Spinner
    ğŸ“œ documentation: âœ”
    ğŸ’¡ examples: âœ˜

Table
    ğŸ“œ documentation: âœ”
    ğŸ’¡ examples: âœ˜

Tbody
    ğŸ“œ documentation: âœ˜
    ğŸ’¡ examples: âœ˜

Td
    ğŸ“œ documentation: âœ˜
    ğŸ’¡ examples: âœ˜

Tfoot
    ğŸ“œ documentation: âœ˜
    ğŸ’¡ examples: âœ˜

Th
    ğŸ“œ documentation: âœ˜
    ğŸ’¡ examples: âœ˜

Thead
    ğŸ“œ documentation: âœ˜
    ğŸ’¡ examples: âœ˜

Tooltip
    ğŸ“œ documentation: âœ”
    ğŸ’¡ examples: âœ˜

Tr
    ğŸ“œ documentation: âœ˜
    ğŸ’¡ examples: âœ˜

Ul
    ğŸ“œ documentation: âœ”
    ğŸ’¡ examples: âœ˜

warning Cannot serialize component properties in:
src/components/alert/alert.jsx

warning Cannot serialize component properties in:
src/components/animate/animate.jsx

warning Cannot serialize component properties in:
src/components/avatar/avatar.jsx

warning Cannot serialize component properties in:
src/components/badge/badge.jsx

warning Cannot serialize component properties in:
src/components/button/button.jsx

warning Cannot serialize component properties in:
src/components/dropdown/dropdown.jsx

warning Cannot parse type of a property 'aria-hidden' in:
src/components/icon/icon.jsx

warning Cannot parse type of a property 'focusable' in:
src/components/icon/icon.jsx

warning Cannot serialize component properties in:
src/components/instrument-badge/instrument-badge.jsx

warning Cannot serialize component properties in:
src/components/labeled-value/labeled-value.jsx

warning Cannot serialize component properties in:
src/components/li/li.jsx

warning Cannot serialize component properties in:
src/components/pane/pane.jsx

warning Cannot serialize component properties in:
src/components/progress-bar/progress-bar.jsx

warning Cannot serialize component properties in:
src/components/spinner/spinner.jsx

warning Cannot serialize component properties in:
src/components/td/td.jsx

warning Cannot serialize component properties in:
src/components/th/th.jsx

warning Cannot serialize component properties in:
src/components/tooltip/tooltip.jsx

"
`;
